File: C++/1052. Grumpy Bookstore Owner.cpp
class Solution {
public:
    int maxSatisfied(vector<int>& customers, vector<int>& grumpy, int X) {
        int len = customers.size();
        int temp = 0;
        for(int i=0;i<len;++i){
            temp += grumpy[i]==0?customers[i]:0;
        }
        for(int i=0;i<X;++i){
            temp += grumpy[i]==1?customers[i]:0;
        }

        int res = temp;
        int l = 0;
        int r = X;

        while(r<len){
            temp -= grumpy[l++]==1?customers[l-1]:0;
            temp += grumpy[r++]==1?customers[r-1]:0;
            res = max(res,temp);
        }
        return res;
    }
};

File: Python3/1052. Grumpy Bookstore Owner.py
class Solution:
    def maxSatisfied(self, customers: List[int], grumpy: List[int], X: int) -> int:
        size = len(customers)
        temp = sum(customers[i] for i in range(size) if grumpy[i]==0)
        
        l = 0
        r = X-1
        temp += sum(customers[i] for i in range(X) if grumpy[i] == 1)
        res = temp

        while r<size-1:
            if grumpy[l] == 1:
                temp -= customers[l]
            l += 1
            r += 1
            if grumpy[r] == 1:
                temp += customers[r]
            res = max(res,temp)
        
        return res
